_pytest/cacheprovider.py

```diff
@@ -168,6 +170,41 @@ def pytest_sessionfinish(self, session):
             config.cache.set("cache/lastfailed", self.lastfailed)
 
 
+class NFPlugin(object):
+    """ Plugin which implements the --nf (run new-first) option """
+
+    def __init__(self, config):
+        self.config = config
+        self.active = config.option.newfirst
+        self.all_items = config.cache.get("cache/allitems", {})
+
+    def pytest_collection_modifyitems(self, session, config, items):
+        if self.active:
+            new_items = []
+            other_items = []
+            for item in items:
+                mod_timestamp = os.path.getmtime(str(item.fspath))
```

Comment:
use `item.fspath.mtime()` here - it works on windows btw, the tests passedthere at least

Reply:
You mean I need rename it in smth like cached_items?